### PR TITLE
TypeScript build

### PR DESCRIPTION
**Pull request by encody**
### 🤔 Why?
- Converting codebase (mostly) to TypeScript

### 🛠 What I changed:
- Added TypeScript-related dev dependencies
- Better project structure
- Most files converted to TypeScript
- Build script for dev & prod
- Dockerfile updates

### 🚦 Functional Testing Results:
- Connected local frontend to render.com PR preview server and it works
- User cards still work

### CODE REVIEW

File: typings.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module '*.png';
```
Comment (Bexanderthebex) [inline typings.d.ts:1]: Will need to delete this as we are not utilizing this in tsconfig.json anymore

File: src/image.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+//   },
+//   {
+//     path: require('../assets/img/badge-stake.png'),
+//     call: badgeStakeSql,
+//   },
+//   {
+//     path: require('../assets/img/badge-deploy.png'),
+//     call: badgeDeploySql,
+//   },
+// ];
```
Comment (Bexanderthebex) [inline src/image.ts:99]: remove commented code?

Comment (encody) [reply src/image.ts:99]: lol whoops

File: typings.d.ts (hunk 1 above)
Comment (encody) [reply typings.d.ts:1]: Should be covered here: https://github.com/NEAR-Edu/stats.gallery-server/pull/18/files#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0R11

Comment (Bexanderthebex) [reply typings.d.ts:1]: ohhhh, that makes sense. alrighty thank you for pointing that out

**Pull request merged**